frontend/test/resolution/testScopeResolve.cpp

```diff
@@ -570,6 +570,134 @@ static void test13() {
   assert(guard.realizeErrors());
 }
 
+// There's specal handling for the rightmost field access. Make sure this
+// special handling properly handles super.
+static void test14() {
+  printf("test14\n");
+  Context ctx;
+  Context* context = &ctx;
+
+  auto path = UniqueString::get(context, "input.chpl");
+  std::string contents = R""""(
+      module A {
+        var x : int;
+        module B {
+          import this.super as C;
+          var y = C.x;
+        }
+      }
+   )"""";
+  setFileText(context, path, contents);
+
+  const ModuleVec& vec = parseToplevel(context, path);
+
+  const Variable* x = findVariable(vec, "x");
+  assert(x);
+  const Variable* y = findVariable(vec, "y");
+  assert(y);
+
+  const ResolvedExpression& reY = scopeResolveIt(context, y->initExpression());
+  assert(reY.toId() == x->id());
+}
+
+
+// Make sure that the dot-expression handling of "this" works in addition
+// to the idenifier-expression handling of "this". Technically this is
+// redundant, but our goal is to issue a warning, not fail to resolve in
+// this case.
+static void test15() {
+  printf("test15\n");
+  Context ctx;
+  Context* context = &ctx;
+
+  auto path = UniqueString::get(context, "input.chpl");
+  // Note this.super.this instead of this.super in the previous test.
+  std::string contents = R""""(
+      module A {
+        var x : int;
+        module B {
+          import this.super.this as C;
+          var y = C.x;
+        }
+      }
+   )"""";
+  setFileText(context, path, contents);
+
+  const ModuleVec& vec = parseToplevel(context, path);
+
+  const Variable* x = findVariable(vec, "x");
+  assert(x);
+  const Variable* y = findVariable(vec, "y");
+  assert(y);
+
+  const ResolvedExpression& reY = scopeResolveIt(context, y->initExpression());
+  assert(reY.toId() == x->id());
+}
+
+// Ensures that a module declared private and then imported renamed is
+// not accessible externally (renaming doesn't change visibility).
+static void test16() {
+  printf("test16\n");
+  Context ctx;
+  Context* context = &ctx;
+
+  auto path = UniqueString::get(context, "input.chpl");
+  std::string contents = R""""(
+      module A {
+        private module B {
+          var x = 1;
+        };
+        public use B as C;
```

Comment:
Should this also be emitting an error? Does it already and we just don't check for it in this test? Just curious.

Reply:
It should, and I don't check. It would be better to make it check, though.

Reply:
That caught an issue, thanks.